src/Documentation/doc/basic/autocomplete.md

```diff
@@ -0,0 +1,88 @@
+# Autocomplete
+
+The BitBurner terminal offers tab-completion, where pressing tab after typing a script name offers suggestions for arguments to pass to the script.
+
+This relies on an exported function named "autocomplete" that is placed _outside_ of main, in the base scope of the script.
+
+```
+  autocomplete(data, args)
+  data (Object) – general data about the game you might want to autocomplete.
+  args (string[]) – current arguments. Minus run script.js
+```
+
+This function must return an array, the contents of which make up the autocomplete options.
+
+A basic example as a complete script;
+
+```javascript
+export function autocomplete(data, args) {
+  return ["argument0", "argument1", "argument2"];
+}
+
```

Comment:
I'd suggest merging these two by using JSDoc. This has the advantage of making a cut-an-pasteable example that will have proper type hinting.
```suggestion
```js
/**
 * @param {AutocompleteData} data - context about the game, useful when autocompleting
 * @param {string[]} args - current arguments, not including "run script.js"
 * @returns {string[]} the array of possible autocomplete options
 */
export function autocomplete(data, args) {
  return ["argument0", "argument1", "argument2"];
}

/** @param {NS} ns */
```